### PR TITLE
Get raw es query

### PR DESCRIPTION
**Pull request by markrsocialnative**

### CODE REVIEW

**Comment by erhathaway** [comment]
@markrsocialnative can you add comments about what this PR does?

**Comment by erhathaway** [comment]
This looks very similar to `runCustomFilterQuery` but achieves the same thing through an internal function. Is the desire is to be able to submit a starting ES query that is reduced over?

Another approach could be to expand the type signature of `runCustomFilterQuery` to include a 'startingESQuery` as an option. The benefit of this is that the public api surface area doesn't grow as much.

File: package.json
Comment (dearsaturn) [inline package.json:3]: Should be 4.5.0

**Review by dearsaturn — APPROVED** [review]
One thing, but otherwise

[LGTM](https://media2.giphy.com/media/ndKWMoPb2jSlG/giphy.gif)

**Comment by dearsaturn** [comment]
@erhathaway The goal is to be able to hand the ES query over to some other context. For our implementation into the legacy app, we need to be able to extract the raw query and send it to an endpoint that will integrate with the lead hopper.

File: src/manager.ts
Comment (erhathaway) [inline src/manager.ts:1059]: does this need to be awaited too? it looks like it was removed

File: src/manager.ts
Comment (erhathaway) [inline src/manager.ts:1004]: can we call this method `generateESRequestFromCurrentFilterState` or something a little more explicit?

**Comment by erhathaway** [comment]
> For our implementation into the legacy app, we need to be able to extract the raw query and send it to an endpoint that will integrate with the lead hopper.

Ah! Ok this looks good then

File: src/manager.ts
Comment (erhathaway) [reply src/manager.ts:1004]: Oh @markrsocialnative can you also document this method in the README?

File: package.json
Comment (markrsocialnative) [reply package.json:3]: branch `master` is at ```    "version": "4.5.0", ``` so it will fail CI if it's not > than that

File: src/manager.ts
Comment (markrsocialnative) [reply src/manager.ts:1123]: I understand
```js
function foo() {
  return Promise.resolve('baz')
}
```
and
```js
async function bar() {
  const baz = await Promise.resolve('baz');
  return baz;
}
```
to be equivalent.

What I just noticed that complicates things is the inclusion of a try / catch block. But it looks like the try / catch does nothing anyhow.
I believe
```js
try {
  foo();
catch (e) {
  throw(e)
}
```
to be equivalent to 
```js
foo();
```

File: src/manager.ts
Comment (markrsocialnative) [reply src/manager.ts:1068]: yep

File: package.json
Comment (dearsaturn) [reply package.json:3]: Gotcha! In that case, maybe you should rebase?

**Review by erhathaway — APPROVED** [review]
🚀

File: src/manager.ts
Comment (dearsaturn) [reply src/manager.ts:1123]: Agreed, I think we should remove the try-catch since it's not doing anything

Comment (erhathaway) [reply src/manager.ts:1123]: Lol i've actually done that in a lot of places. There are random catches everywhere....

**Review by erhathaway — CHANGES_REQUESTED** [review]
Oh wait, i forgot to check if the readme was updated. Still need that to happen before approval

**Review by dearsaturn — DISMISSED** [review]
[LGTM](https://media.gettyimages.com/photos/upset-golfer-looking-at-golf-ball-at-hole-edge-picture-id80509649?s=612x612)

File: README.md
Comment (erhathaway) [inline README.md:452]: Can you also add this to api -> manager -> methods:

[https://github.com/social-native/elastic-composer/blob/master/README.md#methods](https://github.com/social-native/elastic-composer/blob/master/README.md#methods)

**Comment by dearsaturn** [comment]
Bump

**Pull request merged**
